substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/NativeImageGenerator.java

```diff
@@ -367,6 +392,33 @@ public static Platform defaultPlatform(ClassLoader classLoader) {
         }
     }
 
+    /**
+     * Returns String representing OS that platform is based on.
+     */
+    public static String getPlatformOS(Platform p) {
+        /*
+         * This assumes that given platform is extended from PLATFORM_JNI as it is the case with all
+         * default platforms that native-image supports. Better solution would be to define
+         * appropriate interfaces for OS and Architecture which would then be implemented by all
+         * Platform classes.
+         */
+        Class<?>[] interfaces = p.getClass().getInterfaces();
+        for (Class<?> i : interfaces) {
+            if (InternalPlatform.PLATFORM_JNI.class.isAssignableFrom(i)) {
+                return i.getSimpleName();
+            }
+        }
+        VMError.shouldNotReachHere("Couldn't find OS for specified platform. Check if provided Platform class implements PLATFORM_JNI.");
```

Comment:
throw should go here and return is then not necessary.

Reply:
Fixed.